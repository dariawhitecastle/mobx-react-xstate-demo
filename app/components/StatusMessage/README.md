<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->

## createStatusMessage

A StatusMessage

## Usage

```javascript
import { StatusMessage } from 'components/StatusMessage'
<StatusMessage error>Ooops, there was an error!</StatusMessage>
```

## Properties

**error**

Boolean to switch the flag from error to success

```
boolean
```

**children**

Any children to render by the styled message

```
node
```
