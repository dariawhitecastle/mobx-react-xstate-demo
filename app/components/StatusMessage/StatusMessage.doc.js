import { describe, PropTypes } from 'react-desc'

export const doc = StatusMessage => {
  const DocumentedStatusMessage = describe(StatusMessage)
    .description('A StatusMessage')
    .usage(
      `import { StatusMessage } from 'components/StatusMessage'
<StatusMessage error>Ooops, there was an error!</StatusMessage>`,
    )

  DocumentedStatusMessage.propTypes = {
    error: PropTypes.bool
      .description(
        'Boolean to control if the message is displayed as an error (status-error color) or success (status-ok color)',
      )
      .defaultValue(false),
    children: PropTypes.node.description('Any children to render by the styled message'),
  }

  return DocumentedStatusMessage
}
