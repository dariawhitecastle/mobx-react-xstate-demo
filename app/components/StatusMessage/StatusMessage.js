/* eslint-disable react/prop-types */
import React from 'react'

import { doc } from './StatusMessage.doc'
import { Header } from '../Header'
import helpers from '../../utils/helpers'

/**
 *
 * StatusMessage
 *
 * Displays success or error message based on the error prop value
 *
 */
const createStatusMessage = ({ error, children, ...rest }) => (
  <Header level="5" margin="small" color={error ? 'status-error' : 'status-ok'} {...rest}>
    {children}
  </Header>
)

export default helpers.createWithDoc({
  docFunction: doc,
  component: createStatusMessage,
})
