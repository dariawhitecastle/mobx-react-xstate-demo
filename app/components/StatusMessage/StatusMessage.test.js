/**
*
* Tests for StatusMessage
*
* @see
https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
*
*/

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'
// import 'jest-dom/extend-expect'; // add some helpful assertions

import StatusMessage from './StatusMessage'
import { DEFAULT_LOCALE } from '../../i18n'

const renderComponent = (props = {}) => render(<StatusMessage {...props} />)

test('it renders and matches snapshot', () => {
  const { container } = renderComponent()
  expect(container).toMatchSnapshot()
})

test('It does not log to the console', () => {
  const spy = jest.spyOn(global.console, 'log')
  render(
    <IntlProvider locale={DEFAULT_LOCALE}>
      <StatusMessage />
    </IntlProvider>,
  )
  expect(spy).not.toHaveBeenCalled()
})
