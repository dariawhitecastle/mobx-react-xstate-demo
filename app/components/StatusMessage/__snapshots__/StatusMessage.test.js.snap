// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches snapshot 1`] = `
<div>
  <h5
    class="StyledHeading-sc-1rdh4aw-0 ERfVz"
  />
</div>
`;
